connect/transforms/src/main/java/org/apache/kafka/connect/transforms/field/SingleFieldPath.java

```diff
@@ -0,0 +1,500 @@
+                    steps.add(field);
+                    idx++;
+                } else { // add all
+                    String field = pathText.substring(start);
+                    steps.add(field);
+                }
+            } else {
+                StringBuilder field = new StringBuilder();
+                idx++;
+                int start = idx;
+                while (true) {
+                    idx = pathText.indexOf(String.valueOf(BACKTICK), idx);
+                    if (idx == -1) { // if not found, fail
+                        throw new IllegalArgumentException("Incomplete backtick pair in path: " + pathText);
+                    }
+
+                    boolean atEndOfPath = idx >= pathText.length() - 1;
+                    if (atEndOfPath) {
+                        field.append(pathText, start, idx);
+                        // we've reached the end of the path, and the last character is the backtick
+                        steps.add(field.toString());
+                        idx++;
+                        break;
+                    }
+
+                    boolean notFollowedByDot = pathText.charAt(idx + 1) != DOT;
+                    if (notFollowedByDot) {
+                        boolean afterABackslash = pathText.charAt(idx - 1) == BACKSLASH;
+                        if (afterABackslash) {
+                            // this backtick was escaped; include it in the field name, but continue
+                            // looking for an unescaped matching backtick
+                            field.append(pathText, start, idx - 1)
+                                .append(BACKTICK);
+
+                            idx++;
+                            start = idx;
+                        } else {
+                            // this backtick isn't followed by a dot; include it in the field name, but continue
+                            // looking for a matching backtick that is followed by a dot
+                            idx++;
+                        }
+                        continue;
+                    }
+
+                    boolean afterABackslash = pathText.charAt(idx - 1) == BACKSLASH;
+                    if (afterABackslash) {
+                        // this backtick was escaped; include it in the field name, but continue
+                        // looking for an unescaped matching backtick
+                        field.append(pathText, start, idx - 1)
+                            .append(BACKTICK);
+
+                        idx++;
+                        start = idx;
+                        continue;
+                    }
+                    // we've found our matching backtick
+                    field.append(pathText, start, idx);
+                    steps.add(field.toString());
+                    idx += 2; // increment by two to include the backtick and the dot after it
+                    break;
+                }
+
+
```

Comment:
Nit: unnecessary newlines (allows less code to fit on the screen, makes it a little harder to read)
```suggestion
```